### ISSUE TITLE
deposit-ui (and others): modals should auto-focus their inputs

### ISSUE THREAD

**Issue by slint** [inveniosoftware/invenio-app-rdm#2455]
When opening modals in the deposit form, their first (or "best" actionable) input should be auto-focused. Some examples:

- [x] Creators/Contributors "+ Add creator" modal: The focus should be on the search input (i.e. to quickly be able to search for an author, navigate with arrows and click enter to select)
  - [x] Extra: when `Tab`ing through the form, once you reach the "Role" field, it wrongly automatically selects the first option. The only way to clear it afterwards is to click the `X` at the right of the dropdown
- [x] Licenses "+ Add custom" modal: Should focus on "Title" input field
- [x] Awards "+ Add custom" modal: Should focus on "Funder" search dropdown

Some good examples of modals that auto-focus correctly:

- "Select a community" modal at top of the deposit form (focuses on the search input)
- Licenses "+ Add standard" modal (focuses on the search input)
- Awards "+ Add award" modal (focuses on the search input)
- On the record landing page, all the community management modals on the right sidebar (they focus on their search inputs)
- Member invitation modal (focuses on the "People" tab inside the modal)
  - 👉  This could be possibly improved by focusing on the member search input directly (but still much better, since you can reach it with a single `Tab`)

**Comment by jennur** [comment]
Just a comment to clear up some concerns! 

"Member invitation modal (focuses on the "People" tab inside the modal). This could be possibly improved by focusing on the member search input directly (but still much better, since you can reach it with a single Tab)". 

The people tab is intentionally focused on opening because it's the first interactive element in the modal. It should be this way because of accessibility concerns (see https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/) – e.g. screen reader users (who can't see the UI) rely on the tab flow announced by the screen reader as they navigate.

**Comment by slint** [comment]
Thanks, that makes sense :)

I think we'll have to see how much the "Organization" option is used and if it's not we could redesign to make the most common path (i.e. "Person") quicker, and move the "Organization" option to something more secondary.
